### PR TITLE
feat: add brc-20 send for non-zero-index taproot addresses, closes #3830

### LINKED ISSUE #3830: Add BRC-20 transfer support for non-zero-index taproot addresses

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/5325007604).

See title

https://github.com/hirosystems/wallet/issues/3830

### CODE REVIEW

File: src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts [hunk 1]
```diff
@@ -65,3 +76,78 @@ export function useBrc20TokensByAddressQuery<T extends unknown = FetchBrc20Token
     ...options,
   });
 }
+export function useBrc20TokensQuery() {
+  const network = useCurrentNetwork();
+  const keychain = useTaprootCurrentAccountPrivateKeychain();
```
Comment (kyranjamie) [inline src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts]: Would be better to do this from the signer hook and avoid `getTaprootAddress` if possible.

Think should work:

```ts
const createSigner = useCurrentAccountTaprootSigner();
const address = createSigner(index).address;
```

File: src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts [hunk 2]
```diff
@@ -53,15 +61,65 @@ async function fetchBrc20TokensByAddress(address: string): Promise<Brc20Token[]>
...
-    ...options,
+      return {
+        brc20Tokens,
+        fromIndex,
+      };
+    },
+    getNextPageParam(prevInscriptionQuery) {
+      const { fromIndex, brc20Tokens } = prevInscriptionQuery;
+
+      if (fromIndex >= stopSearchAfterNumberAddressesWithoutBrc20Tokens) {
```
Comment (alter-eggo) [inline src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts]: after ~25-30 requests api starts to resp with 429 error, so prob better to stop just after 20 (or less?)

File: src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts [hunk 3]
```diff
@@ -53,15 +61,65 @@ async function fetchBrc20TokensByAddress(address: string): Promise<Brc20Token[]>
...
+  const network = useCurrentNetwork();
+  const nativeSegwitSigner = useCurrentAccountNativeSegwitIndexZeroSigner();
+  const currentBitcoinAddress = nativeSegwitSigner.address;
+  const createSigner = useCurrentAccountTaprootSigner();
+
+  const getTaprootAddressData = useCallback(
+    (fromIndex: number, toIndex: number) => {
+      return createNumArrayOfRange(fromIndex, toIndex - 1).map(num => {
+        // TO-DO remove this check when we have a better way to handle this
+        if (!createSigner) return '';
```
Comment (kyranjamie) [inline src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts]: If you `if (!createSigner) throw new Error('No signer');` outside this fn, do you still need the check?

File: src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts [hunk 4]
```diff
@@ -53,15 +61,65 @@ async function fetchBrc20TokensByAddress(address: string): Promise<Brc20Token[]>
...
 }
 
-type FetchBrc20TokensByAddressResp = Awaited<ReturnType<typeof fetchBrc20TokensByAddress>>;
+export function useBrc20TokensQuery() {
+  const network = useCurrentNetwork();
+  const nativeSegwitSigner = useCurrentAccountNativeSegwitIndexZeroSigner();
+  const currentBitcoinAddress = nativeSegwitSigner.address;
+  const createSigner = useCurrentAccountTaprootSigner();
+
+  const getTaprootAddressData = useCallback(
```
Comment (kyranjamie) [inline src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts]:
```suggestion
  const getNextTaprootAddressBatch = useCallback(
```

Suggestion for more readable name?

or 
image

To me "data" implies that there's some metadata beyond the addresses themselves

File: src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts (hunk 2 above)
Comment (kyranjamie) [reply src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts]: Let's check for 5 empty addresses before stopping

**Review by kyranjamie — APPROVED** [review]

File: src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts (hunk 2 above)
Comment (alter-eggo) [reply src/app/query/bitcoin/ordinals/brc20/brc20-tokens.query.ts]: done

**Review by kyranjamie — APPROVED** [review]

**Pull request merged**